nixos/modules/system/boot/systemd-nspawn.nix

```diff
@@ -126,7 +153,7 @@ in {
           systemd.services."systemd-nspawn@".serviceConfig.ExecStart = [ 
             ""  # deliberately empty. signals systemd to override the ExecStart
             # Only difference between upstream is that we do not pass the -U flag
-            "${config.systemd.package}/bin/systemd-nspawn --quiet --keep-unit --boot --link-journal=try-guest --network-veth --settings=override --machine=%i"
+            "${config.systemd.package}/bin/systemd-nspawn --quiet --boot --link-journal=try-guest --network-veth --settings=override --machine=%i"
```

Comment:
Removing `--keep-unit` here shouldn't be necessary here anymore since v245 given you already have `KillSignal = "SIGRTMIN+3";` .  We really ought to test if these issues are fixed in 245 and otherwise update the upstream systemd ticket

Reply:
This is blocked on the systemd bump, which is blocked on some of the tests failing currently.

Reply:
@Ma27 can you check this again now that systemd 245.3 has been merged to staging?